### PR TITLE
darwin: work around condition variable kernel bug

### PR DESCRIPTION
**Pull request by bnoordhuis**
It has been reported that destroying condition variables that have been
signalled but not waited on can sometimes result in application crashes.
See https://codereview.chromium.org/1323293005.

CI: https://ci.nodejs.org/job/libuv-test-commit/41/

### CODE REVIEW

**Comment by saghul** [comment]
LGTM

File: src/unix/thread.c [hunk 1]
```diff
@@ -392,6 +392,35 @@ int uv_cond_init(uv_cond_t* cond) {
...
+  if (pthread_mutex_init(&mutex, NULL))
+    abort();
+
+  if (pthread_mutex_lock(&mutex))
+    abort();
+
+  ts.tv_sec = 0;
+  ts.tv_nsec = 1;
+
+  err = pthread_cond_timedwait_relative_np(cond, &mutex, &ts);
```
Comment (kthelgason) [inline src/unix/thread.c]: Doesn't this call release the mutex, making the `unlock` below redundant?

Comment (bnoordhuis) [inline src/unix/thread.c]: No.  The mutex is released while in the call but is acquired again on return.

Comment (kthelgason) [inline src/unix/thread.c]: Of course, sorry. Returning to the caller with the mutex unlocked would make no sense. I was going off some code I saw elsewhere, should have read the manual.

**Pull request merged**